com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -826,6 +826,7 @@ internal static void WriteNetworkVariableData(List<INetworkVariable> networkVari
                         else
                         {
                             networkVariableList[j].WriteField(stream);
+                            writer.WritePadBits();
```

Comment:
Interesting.  Did we find a bug here?   Or is there an encoding change?

Reply:
So the issue here was not what I initially thought... it was that NetworkList was writing 3 bits, and when there's an unaligned write left at the end of the buffer, the Position value ends up being one byte too small to get those last 3 bits. This just makes sure the buffer is aligned after the write.

There's no specific case in WriteField that's doing this right now, but my figuring is, if the bug has happened in WriteDelta, we should future-proof WriteField against it, too... otherwise WriteField is a landmine waiting to explode some day.

Reply:
I really like on our offline convo about just banishing bitwise writing (or having a wrapper which lets you bitwise write but then it pads it for you) to prevent more weird fragile cases like this and because we've learned from other netcode folks the wisdom of not being over-the-top-miserly on space.  In the meantime, if this shores us up, great